workers/unity/Assets/Fps/Scripts/WorldTiles/MapTemplate.cs

```diff
@@ -0,0 +1,113 @@
+                if (index.a == 0xff)
+                {
+                    if (!tileLookup.TryGetValue(index, out collection))
+                    {
+                        Debug.Log($"Unknown color: {index}\nThere are {tileLookup.Count} colors available.");
+                        PrintColors();
+                        collection = defaultTileCollection;
+                    }
+                }
+            }
+
+            return collection.GetRandomTile(random);
+        }
+
+        private void OnEnable()
+        {
+            InitializeLookup();
+        }
+
+        private void PrintColors()
+        {
+            var message = tileLookup.Select(element => $"[{element.Key}]")
+                .Aggregate("", (current, next) => $"{current} {next}");
+            Debug.Log(message);
+        }
+
+        private void OnValidate()
+        {
+            // Check for issues
+            if (!templateBitmap.isReadable)
+            {
+                Debug.LogError($"template bitmap {templateBitmap.name} must be Read/Write enabled.");
+            }
+
+            if (templateBitmap.format != TextureFormat.RGBA32)
+            {
+                Debug.LogError($"template bitmap {templateBitmap.name} must be RGBA32.");
+            }
+
+            if (defaultTileCollection == null)
+            {
+                Debug.LogError($"Default Tile Collection must be set.");
+            }
+
+            if (unitSize == 0)
+            {
+                unitSize = 1;
+            }
+
+            InitializeLookup();
+        }
+
+        private void InitializeLookup()
+        {
+            // Fill lookup map
+            tileLookup.Clear();
+            if (tileCollections != null && tileCollections.Length > 0)
+            {
+                foreach (var tileType in tileCollections)
+                {
+                    if (tileType == null)
+                    {
+                        continue;
+                    }
+
+                    var index = ColorToLookup(tileType.DisplayColor);
+                    if (tileLookup.ContainsKey(index))
+                    {
+                        Debug.LogError($"Tile collection with color {ColorUtility.ToHtmlStringRGBA(tileType.DisplayColor)} already exists.");
+                        continue;
+                    }
+
+                    tileLookup.Add(index, tileType);
+                }
+            }
+        }
+
+        private Color32 ColorToLookup(Color color)
+        {
+            return new Color32((byte) Mathf.Clamp(Mathf.RoundToInt(color.r * (float) byte.MaxValue), 0, (int) byte.MaxValue), (byte) Mathf.Clamp(Mathf.RoundToInt(color.g * (float) byte.MaxValue), 0, (int) byte.MaxValue), (byte) Mathf.Clamp(Mathf.RoundToInt(color.b * (float) byte.MaxValue), 0, (int) byte.MaxValue), (byte) Mathf.Clamp(Mathf.RoundToInt(color.a * (float) byte.MaxValue), 0, (int) byte.MaxValue));
```

Comment:
OMG
I have literally no idea what this does

Reply:
Takes every color element, from 0 to 1, and multiplies it by 255 and clamps it to 0 -> 255